polkadot/xcm/xcm-executor/src/assets.rs

```diff
@@ -350,25 +362,27 @@ impl AssetsInHolding {
 				for asset in assets.into_inner().into_iter() {
 					match asset {
 						Asset { fun: Fungible(amount), id } => {
-							let (remove, amount) = match self.fungible.get_mut(&id) {
+							let (remove, balance) = match self.fungible.get_mut(&id) {
 								Some(self_amount) => {
-									let amount = amount.min(*self_amount);
-									*self_amount -= amount;
-									(*self_amount == 0, amount)
+									// Ok to use `saturating_take()` because we checked with
+									// `self.ensure_contains()` above against `saturate` flag
+									let balance = self_amount.saturating_take(amount);
```

Comment:
if `amount` < `self_amount`, this wont fail and will remove the `self_amount`. not sure if this an issue.